x-pack/platform/plugins/shared/streams/server/lib/telemetry/stats/stats_telemetry_service.ts

```diff
@@ -0,0 +1,228 @@
+import { AssetClient } from '../../streams/assets/asset_client';
+import { SystemClient } from '../../streams/system/system_client';
+import { QueryClient } from '../../streams/assets/query/query_client';
+import { assetStorageSettings } from '../../streams/assets/storage_settings';
+import { systemStorageSettings } from '../../streams/system/storage_settings';
+import { streamsStorageSettings } from '../../streams/service';
+import { migrateOnRead } from '../../streams/helpers/migrate_on_read';
+import { registerStreamsUsageCollector } from './streams_usage_collector';
+
+/**
+ * Service for collecting Streams usage statistics for telemetry
+ *
+ * This service creates internal StreamsClient instances optimized for read-only
+ * telemetry collection. It bypasses authentication requirements by using:
+ *
+ * 1. Internal Elasticsearch clients (asInternalUser) for direct data access
+ * 2. Internal SavedObjects repositories for system-level access
+ * 3. Read-only mock RulesClients that handle interface requirements without auth
+ */
+export class StatsTelemetryService {
+  private readonly logger: Logger;
+  private readonly isDev: boolean;
+
+  constructor(logger: Logger, isDev: boolean) {
+    this.logger = logger;
+    this.isDev = isDev;
+  }
+
+  public setup(
+    core: CoreSetup<StreamsPluginStartDependencies>,
+    usageCollection?: UsageCollectionSetup
+  ) {
+    if (usageCollection) {
+      this.logger.debug('[Streams Stats Telemetry Service] Setting up streams usage collector');
+      registerStreamsUsageCollector(
+        usageCollection,
+        this.logger,
+        () => this.getInternalStreamsClient(core),
+        this.isDev
+      );
+    } else {
+      this.logger.debug(
+        '[Streams Stats Telemetry Service] Usage collection not available, skipping setup'
+      );
+    }
+  }
+
+  /**
+   * Creates an internal StreamsClient for telemetry collection
+   * Uses minimal clients that bypass/faking authentication requirements
+   */
+  private async getInternalStreamsClient(
+    core: CoreSetup<StreamsPluginStartDependencies>
+  ): Promise<StreamsClient> {
+    const [coreStart] = await core.getStartServices();
+    const logger = this.logger;
+
+    // Create minimal clients for telemetry that don't require authentication
+    const assetClient = await this.createMinimalAssetClient(core);
+    const systemClient = await this.createMinimalSystemClient(core);
+    const queryClient = await this.createMinimalQueryClient(core, assetClient);
+
+    // Create StreamsClient with internal cluster client
+    const internalClusterClient = coreStart.elasticsearch.client.asInternalUser;
+    const isServerless = coreStart.elasticsearch.getCapabilities().serverless;
+
+    // Create a mock scoped cluster client that uses internal user
+    const mockScopedClusterClient = {
+      asCurrentUser: internalClusterClient,
+      asInternalUser: internalClusterClient,
+    };
+
+    const storageAdapter = new StorageIndexAdapter(
```

Comment:
Not a big fan of having the initialization logic in a two places now, can we refactor so there is just one place that creates this? As mentioned in the other comment, this part should be fairly clean.